xbmc/epg/EpgSearchFilter.h

```diff
@@ -20,42 +20,107 @@
-
-    virtual bool MatchGenre(const CEpgInfoTag &tag) const;
-    virtual bool MatchDuration(const CEpgInfoTag &tag) const;
-    virtual bool MatchStartAndEndTimes(const CEpgInfoTag &tag) const;
-    virtual bool MatchSearchTerm(const CEpgInfoTag &tag) const;
-    virtual bool MatchChannelNumber(const CEpgInfoTag &tag) const;
-    virtual bool MatchChannelGroup(const CEpgInfoTag &tag) const;
-    virtual bool MatchBroadcastId(const CEpgInfoTag &tag) const;
-    virtual bool MatchChannelType(const CEpgInfoTag &tag) const;
+    bool FilterEntry(const CEpgInfoTagPtr &tag) const;
 
+    /*!
+     * @brief remove duplicates from a list of epg tags.
+     * @param results the list of epg tags.
+     * @return the number of items in the list after removing duplicates.
+     */
     static int RemoveDuplicates(CFileItemList &results);
 
+    const std::string &GetSearchTerm() const { return m_strSearchTerm; }
+    void SetSearchTerm(const std::string &strSearchTerm) {m_strSearchTerm = strSearchTerm; }
+
+    bool IsCaseSensitive() const { return m_bIsCaseSensitive; }
+    void SetCaseSensitive(bool bIsCaseSensitive) { m_bIsCaseSensitive = bIsCaseSensitive; }
+
+    bool ShouldSearchInDescription() const { return m_bSearchInDescription; }
+    void SetSearchInDescription(bool bSearchInDescription) {m_bSearchInDescription = bSearchInDescription; }
+
+    int GetGenreType() const { return m_iGenreType; }
+    void SetGenreType(int iGenreType) { m_iGenreType = iGenreType; }
+
+    int GetGenreSubType() const { return m_iGenreSubType; }
+    void SetGenreSubType(int iGenreSubType) { m_iGenreSubType = iGenreSubType; }
+
+    int GetMinimumDuration() const { return m_iMinimumDuration; }
+    void SetMinimumDuration(int iMinimumDuration) { m_iMinimumDuration = iMinimumDuration; }
+
+    int GetMaximumDuration() const { return m_iMaximumDuration; }
+    void SetMaximumDuration(int iMaximumDuration) { m_iMaximumDuration = iMaximumDuration; }
+
+    const CDateTime &GetStartDateTime() const { return m_startDateTime; }
+    void SetStartDateTime(const CDateTime &startDateTime) { m_startDateTime = startDateTime; }
+
+    const CDateTime &GetEndDateTime() const { return m_endDateTime; }
+    void SetEndDateTime(const CDateTime &endDateTime) { m_endDateTime = endDateTime; }
+
+    bool ShouldIncludeUnknownGenres() const { return m_bIncludeUnknownGenres; }
+    void SetIncludeUnknownGenres(bool bIncludeUnknownGenres) { m_bIncludeUnknownGenres = bIncludeUnknownGenres; }
+
+    bool ShouldRemoveDuplicates() const { return m_bRemoveDuplicates; }
+    void SetRemoveDuplicates(bool bRemoveDuplicates) { m_bRemoveDuplicates = bRemoveDuplicates; }
+
+    bool IsRadio() const { return m_bIsRadio; }
+    void SetIsRadio(bool bIsRadio) { m_bIsRadio = bIsRadio; }
+
+    int GetChannelNumber() const { return m_iChannelNumber; }
+    void SetChannelNumber(int iChannelNumber) { m_iChannelNumber = iChannelNumber; }
+
+    bool IsFTAOnly() const { return m_bFTAOnly; }
```

Comment:
Since you've used `MatchFreeToAir`, maybe these should be `FreeToAir` too? Including the member variable.